clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -4682,6 +4682,7 @@ bool Sema::CheckCallingConvAttr(const ParsedAttr &Attrs, CallingConv &CC,
 
   TargetInfo::CallingConvCheckResult A = TargetInfo::CCCR_OK;
   const TargetInfo &TI = Context.getTargetInfo();
+  bool suppress_sycl_cconv_warning = false;
```

Comment:
I don't think any of this is necessary... I think the CC=CC_C; above (4619) just needs to be CC= TI.getDefaultCallingConv();

Reply:
But what if it is an invalid (but not bogus) calling convention like fastcall?   Shouldn't we still warn?

Reply:
Why wouldn't that get caught in the normal fastcall control flow?

Reply:
Where is that?  I don't see it getting caught elsewhere.

Reply:
we already prohibit __fastcall, right?  Why wouldn't it still get caught that way?

Reply:
That's what I am saying.   I don't see us prohibiting these calling conventions anywhere else.  We warn and change it to the "default" here.

Reply:
Doesn't the CCCR_Warning branch already do that?  The !CUDA branch above does: 
    } else {
        A = TI.checkCallingConvention(CC);
      }

If cdecl will have caused CC == CC_SPIRV (or however it is spelled) since it is the default.  Anything else returns CCCR_Warning.

The switch statement below will warn, then set CC to the default calling convention.  This branch should be as close to https://github.com/llvm/llvm-project/blob/master/clang/lib/Sema/SemaDeclAttr.cpp#L4611 as possible...

Reply:
Ah, got it.  Initially I thought you meant to also get rid of the changes in the Target files.
Will upload shortly.

Reply:
Okay :-), one more question.   Should the like at ~4619 be something like:
CC = LangOpts.SYCLIsDevice ? TI.getDefaultCallingConv() : CC_C;
or is it safe to always change CC_C to the default?

Reply:
Its safe to do it ALWAYS.  Every other TargetInfo does CC_C as the default.